gcc/rust/typecheck/rust-hir-dot-operator.h

```diff
@@ -35,6 +35,13 @@ struct MethodCandidate
   }
 
   bool is_error () const { return candidate.is_error (); }
+
+  DefId get_defid () const { return candidate.get_defid (); }
+
+  bool operator<(const MethodCandidate &c) const
```

Comment:
Argh :D Operator overloading! Blasphemy! 

```suggestion
  bool less_than(const MethodCandidate &c) const
```

So that 
```c++
method_candidate1 < method_candidate2;
```

becomes
```c++
method_candidate1.less_than(method_candidate2);
```

Is this for a `std::hashmap` or `std::set` or something and I'm not realizing?

Reply:
Yeah it seems if we implement operator< std::set works for us is why I implemented it. I really wish STL was more sane than this...